unittests/APITests/ReentrantMutex.cpp

```diff
@@ -0,0 +1,154 @@
+#include <catch2/catch.hpp>
+#include <chrono>
+#include <csetjmp>
+#include <FEXCore/Utils/ReentrantMutex.h>
+#include <thread>
+#include <signal.h>
+
+// Test that ensures the Reentrant mutex will timeout without signaling
+TEST_CASE("SimpleWait", "[Reentrant]") {
+  auto Dur = std::chrono::seconds(1);
+  FEXCore::ReentrantMutex Mutex{};
+
+  auto Now = std::chrono::high_resolution_clock::now();
+  bool Signaled = Mutex.WaitFor(Dur);
+  auto End = std::chrono::high_resolution_clock::now();
+
+  // We weren't signaled
+  REQUIRE(Signaled == false);
+
+  // We waited at least the full duration
+  REQUIRE((End - Now) >= Dur);
+}
+
+void WaitThread(FEXCore::ReentrantMutex *Mutex, bool *Signaled) {
+  auto Dur = std::chrono::seconds(5);
+  *Signaled = Mutex->WaitFor(Dur);
+}
+
+// Test that ensure the Reentrant mutex will signal without timing out
+TEST_CASE("SignaledWait", "[Reentrant]") {
```

Comment:
This is correct syntax now, but I think this tag should be dropped entirely in favor of giving the tests a subcategory name like `SignaledWait.ReentrantMutex.APITest`, which would allow for filtering out relevant tests too.

Reply:
Tag dropped